routes/settings.js

```diff
@@ -39,7 +39,14 @@ router.post('/template', async (req, res) => {
 
 router.post('/', async (req, res) => {
     const formData = req.body;
-    formData['ghosler.url'] = req.protocol + '://' + req.hostname;
+
+    let fullUrl = new URL(`${req.protocol}://${req.get('Host')}${req.originalUrl}`);
+    if (req.get('Referer')) {
+        fullUrl = new URL(req.get('Referer'));
+    }
+    fullUrl.pathname = fullUrl.pathname.split('/settings')[0];
+    fullUrl.search = ''; // drops any query parameters
+    formData['ghosler.url'] = fullUrl.toString();
```

Comment:
Ig we also need to remove the last `/` if it exists.
This might cause `404`s due to urls being constructed as 'https://xyz.com/ghosler//settings'.